### PR TITLE
Fix the libdarknet_mac2.0.so path error in from_darknet.py

### PR DESCRIPTION
**Pull request by lkshore**
This change list is intended to fix the `libdarknet_mac2.0.so` path error in `tutorials/frontend/from_darknet.py`, which is raised in [topic 1](https://discuss.tvm.ai/t/end-to-end-tvm-vta-flow-tutorial-with-yolo-v3/3542/13) and [topic 2](https://discuss.tvm.ai/t/runtimeerror-cannot-dereference-null-pointer-from-cdata-float/71) 

`from_darknet.py` has been tested on my mac using commit c3f02c4b81a28c4d9a9a9682635499df651f3d3b and it passed successfully.

### CODE REVIEW

**Comment by tmoreau89** [comment]
Thanks for the fix. I think it would be better to submit a PR to upload the library to the `https://github.com/dmlc/web-data/tree/master/darknet/lib` repo path; and therefore rely upon the same `REPO_URL` rather than getting libraries from two different repos. Do you mind uploading the library first to  `web-data` and updating this PR with the new path?

**Comment by lkshore** [comment]
@tmoreau89 

Sure I can do that. At first I was thinking upload the `libdarknet_mac2.0.so` file as well so we can use the same `REPO_URL`, but my concern is the copyright of that `.so` belongs to the author in the author's own github repo.

I know he is also in TVM community, but do we need to let him know before uploading the `.so`?

-Hao

**Comment by tmoreau89** [comment]
Thanks, I understand. Let's check with @siju-samuel ! Hope you're okay with @lkshore uploading the `libdarknet_mac2.0.so` from your repo to the `dmlc/web-data` repo! That'll fix the darknet tutorial on mac.

**Comment by siju-samuel** [comment]
@tmoreau89 no issues. @lkshore please go ahead and upload the .so built for Mac. Thanks.

**Comment by lkshore** [comment]
The PR of uploading `libdarknet_mac2.0.so` is posted [here](https://github.com/dmlc/web-data/pull/211).

After you merge the PR above, there is no need to modify `from_darknet.py`. you can close this PR. Thanks for everybody's help! :)

-Hao

**Comment by tmoreau89** [comment]
Thanks will merge that one in once the PR in the other repo is merged.

File: tutorials/frontend/from_darknet.py [hunk 1]
```diff
@@ -72,7 +72,7 @@
     DARKNET_URL = REPO_URL + 'lib/' + DARKNET_LIB + '?raw=true'
 elif sys.platform == 'darwin':
     DARKNET_LIB = 'libdarknet_mac2.0.so'
-    DARKNET_URL = REPO_URL + 'lib_osx/' + DARKNET_LIB + '?raw=true'
+    DARKNET_URL = 'https://github.com/siju-samuel/darknet/blob/master/lib_osx/' + DARKNET_LIB + '?raw=true'
```
Comment (tmoreau89) [inline tutorials/frontend/from_darknet.py]: Can you now update it to point to REPO_URL, and test that it all works fine? Thanks

**Comment by lkshore** [comment]
`libdarknet_mac2.0.so` is uploaded in this [PR](https://github.com/dmlc/web-data/pull/211).

After the shared library is uploaded to the correct path in `from_darknet.py`, there is no need to modify `from_darknet.py`, so I restored this file. There is no file changed in this PR.

-Hao

File: tutorials/frontend/from_darknet.py (hunk 1 above)
Comment (lkshore) [reply tutorials/frontend/from_darknet.py]: Hi @tmoreau89 , I pulled the lasted code from TVM master, changed the `REPO_URL` back in `libdarknet_mac2.0.so`, and tested again on my local machine. This darknet tutorial runs without issues.

Before you merge and close this PR, do I need to show you the result of my local run? or do I need to do more tests? If not, I think you can merge and close this PR now.

This is my first time contributing to TVM. Thank you so much for all the help!

-Hao

**Comment by tmoreau89** [comment]
Thanks, indeed the fix was applied outside of this repo; and results in no changes necessary on this one. I will consequently close this PR.

**Pull request closed without merging**